### PR TITLE
Initial logging related guideline.

### PR DESCRIPTION
**Pull request by asemt**
Some initial thoughts on logging.
Will be extended in the future.

### CODE REVIEW

File: infra/logging/Logging.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+## {{ book.must }} Make sure the log messages are in a processable format.
+
+A _processable format_ means that the log messages must conform to following rules:
+
+- The log message **must** be valid JSON,
+- the JSON **must** contain a field `@timestamp`,
+- the value of the `@timestamp` field **must** be in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601)
+  - an example of a _valid_ `@timestamp` value would be: `2017-01-30T08:56:47+00:00`
+- the JSON **should** contain a field `@hostname` containing the hostname the log message was produced on,
+- the JSON **should** contain a field "@service" indicating the service/application the log messages belongs to,
```
Comment (nik-kor) [inline infra/logging/Logging.md]: \` instead of double quotes

File: infra/logging/Logging.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+The JSON will always contain a field `_type` containing the environment name as defined via Ansible's `environment_name` variable (part of automation repository). This value is automatically injected by the `filebeat` agent as part of the logging pipeline.
+
+### Payload:
+
+In general, the author of the log message is in full control of the additional payload of the log message. So all information considered helpful, important, necessary for whatever purposes (e.g. debugging, error tracking, metric extraction) should go into the log message. The content of the log message will _not be altered or modified in any way_ by the logging pipeline.
+
+### Storage:
+
+Each succesfully processed log message ends up in two different storage systems (at the same time, so the the log message gets duplicated for redundancy reasons):
```
Comment (nik-kor) [inline infra/logging/Logging.md]: typo - extra `the`

Comment (asemt) [inline infra/logging/Logging.md]: Could you please point me to the "extra the" here?

File: infra/logging/Logging.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+Let's say a log message's JSON payload contais a field `@serviceError` which holds a string value. ES will use this information in the dynamically created schema as manifested inside ES to define that the schema has a key `@serviceError` which has the _type_ `string`. Now, for whatever reasons, another log message, later on feeded into ES, contains again the same JSON field `@serviceError`, this time however containing an _array_ of strings as value. As soon as ES receives this JSON payload, the internal dynamically created schema **will break** because of the changed type of an already existing schema key. As a consequence, this will trigger a mapping exception inside ES, preventing the log message to be stored at all in ES. Ultimately the log message is _lost_ due to the parsing exception. The log message will, however, still be stored in the S3 bucket (the second storage destination).
```
Comment (robertke) [inline infra/logging/Logging.md]: One more goes to typo `JSON payload contais a field` - `contais`, instead `contains`

**Comment by nik-kor** [comment]
+1

**Comment by italolelis** [comment]
👍

**Pull request merged**